cloudbuild/cloudbuild.yaml

```diff
@@ -25,6 +25,7 @@ steps:
     entrypoint: 'bash'
     args: ['/hadoop-connectors/cloudbuild/presubmit.sh', 'unittest']
     env:
+      - 'GCS_TEST_PROJECT_ID=$PROJECT_ID'
```

Comment:
It was required for one of the unit-test. Just for sanity I can provide a dummy value against unit-test projectId. It's just should not be `null`

Reply:
How it run before w/o this var then?

Reply:
This was working earlier as we use to have a [separate function](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTestBase.java#L61) to override the values of projectId. I got rid of that as I have to introduce another flavour of `loadConfig` with `clientType` support and it wasn't scaling well with all variations which were there.

Adding a env variable even for unit-test made sense instead of overriding them for a individual test. I understand it may lead to actual calls getting made to gcs but we can avoid that by providign a dummy value here.

Reply:
I think that this is suboptimal, as we end up requiring to set this var when we run unit tests in IDEs, etc. Is there an other way to fix it w/o requiring this var for init tests?

This made me thing that maybe we can parametrize tests with client type using env var instead of hacking this in code? I.e. we will need to run test twice with 2 different env vars to test 2 different client types, but we will not need to change almost any code, WDYT?

Reply:
> I think that this is suboptimal, as we end up requiring to set this var when we run unit tests in IDEs, etc. Is there an other way to fix it w/o requiring this var for init tests?

Will investigate.

> This made me thing that maybe we can parametrize tests with client type using env var instead of hacking this in code? I.e. we will need to run test twice with 2 different env vars to test 2 different client types, but we will not need to change almost any code, WDYT?

1. Adding a filesystem Configuration variable as env variable seems a little odd.
2. Both the client don't have exact parity on the level test are written for eg. with HTTP client we can have interceptor support but same support is not there with java-storage (or even a GAPIC level). We are relying heavily on interceptors to trace requests during testing.
3. Overall test execution time will increase for sure as it will run those test too which wouldn't depend on `ClientType`. For 2.2.x we already have two flavour `hadoop2` and `hadoop3`. Introducing `ClientType` will make total runs to be 4.

I will still prefer `Client type` to be part of parametrization and applicable only to relevant test.